core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2525,6 +2529,29 @@ private RelBuilder aggregate_(GroupKeyImpl groupKey,
     return project(projects.transform((i, name) -> aliasMaybe(field(i), name)));
   }
 
+  /**
+   * Removed redundant distinct if an input is already unique.
```

Comment:
I would document that this is specifically for aggregates.
And perhaps a better function name would be removeRedundantAggregateDistinct.

Reply:
Renamed to `removeRedundantAggregateDistinct`.